src/zmx2batoid/zmx_parsers.py

```diff
@@ -392,100 +418,440 @@ def extract_pupils(self):
         self.rotation = rotation
         self.offset = offset
         self.tilt = tilt
         self.comment = comment
 
     def __repr__(self):
-        return (
-            f"SurfacePrescriptionData(\n"
-            f"  Rotation Matrix:\n{self.rotation if self.rotation is not None else 'None'}\n"
-            f"  Offset Vector: {self.offset if self.offset is not None else 'None'}\n"
-            f"  Tilt Vector: {self.tilt if self.tilt is not None else 'None'}\n"
-            f"  Comment: {self.comment}\n"
-        )
-
-
-def build_yaml_file(input_file, output_file=None, encoding="utf-16"):
-    """Builds a yaml file with the critical information of the optical design."""
-
-    telescope = ZemaxFileParser(input_file, encoding=encoding)
-
-    data = {}  # Dict that will be dumped to a YAML file
-
-    details = telescope.system_details
-    for key in vars(details):
-        # Convert numpy arrays to lists to enable writing to YAML
-        if type(vars(details)[key]) == np.ndarray:
-            vars(details)[key] = (vars(details)[key]).tolist()
+        return (f"SurfaceCoordinates(\n"
+                f"  Rotation Matrix:\n{self.rotation if self.rotation is not None else 'None'}\n"
+                f"  Offset Vector: {self.offset if self.offset is not None else 'None'}\n"
+                f"  Tilt Vector: {self.tilt if self.tilt is not None else 'None'}\n"
+                f"  Comment: {self.comment}\n")
 
-        data[key] = vars(details)[key]
-
-    surface_names = telescope.get_surface_names()
-
-    for i, surf in enumerate(surface_names):
-        # Now loop over all the attributes in the surface class
-        # (which is a dictionary via vars)
-        # and check for YAML issues
-        for key in vars(telescope.surfaces[surf[0]]):
-            # Convert numpy arrays to lists
-            if type(vars(telescope.surfaces[surf[0]])[key]) == np.ndarray:
-                vars(telescope.surfaces[surf[0]])[key] = (vars(telescope.surfaces[surf[0]])[key]).tolist()
+class SurfacePRD:
+    def __init__(self, name):
+        """
+        Initialize a Surface object with a given name.
+        """
+        self.name = name
 
-        data[f"surface_{i}"] = vars(telescope.surfaces[surf[0]])
+    def __setattr__(self, key, value):
+        """
+        Override attribute setting to clean up specific values like CURV, DIAM, DISZ and PARM.
+        """
+        super().__setattr__(key, value)
+    
+    def __repr__(self):
+        """
+        Represent the Surface object with its name and attributes.
+        """
+        return f"Surface({self.__dict__})"
+    
+    def items(self):
+        return self.__dict__.items()
+    
+    
 
-    # Write to file
-    with open(output_file, mode="wt", encoding="utf-8") as file:
-        yaml.safe_dump(data, file, sort_keys=False)
 
-    return
 
+    
 
 ###################
 ###################
 ###################
+ 
+def main():
+    import numpy as np
```

Comment:
this isn't doing anything so I'm confused why it is here? You don't need a main function for this, do you?

Reply:
it's because I like to keep:
if __name__=='__main__':
    main()
    
   like in C. But here def main() is useless so I just put something random to keep it "alive" 
```suggestion
    pass
```